### DISCUSSION TITLE
Оптимизация загрузки js корзины на сайте.

### DISCUSSION THREAD

**Discussion by theelementwork** [cdek-it/wordpress#62] [Ideas]
1. В стандартной версии плагина при активации плагина очень сильно начинает тормозить (лагать) добавление в корзину и удаление из нее. Удалось модифицировать файл cdek.php путем добавления до 22 строкой кода:
`if(!((isset($_GET['wc-ajax']) && ($_GET['wc-ajax'] == 'add_to_cart' || $_GET['wc-ajax'] == 'remove_from_cart')) || (isset($_GET['action']) && ($_GET['action'] == 'woodmart_update_cart_item' || $_GET['action'] == 'woodmart_ajax_add_to_cart')) || isset($_GET['key']))){
` и закрытием скобки` }` в конце файла. Файл целиком прикрепляю. 
[cdek.zip](https://github.com/user-attachments/files/17850243/cdek.zip)
После добавления кода работа с аякс корзиной значительно ускоряется и работает с той же скоростью как и без плагина. Возможно следует добавить эту доработку вам. Так как мне после каждого обновления приходится это добавлять в ручную.

2. Так как плагин создан преимущественно для русскоязычной аудитории возможно есть смысл перевести текст ошибок на русский язык? Например:
В файле: /wp-content/plugins/cdekdelivery/build/cdek-checkout-map.js

CDEK was unable to load the list of available pickup points, please select another delivery method
на
СДЭК не удалось загрузить список доступных пунктов самовывоза, выберите другой способ доставки.

и

There are no CDEK pick-up points available in this direction, please select another delivery method
на
Ошибка в названии населенного пункта или по данному направлению нет самовывоза СДЭК, выберите другой способ доставки.

Надеюсь это поможет развитию плагина.

**Comment by vermorag** [comment]
Добрый день.
1. Вызов расчета в корзине является стандартным поведением в woocommerce. Ваш вариант не подходит по двум причинам: он не учитывает новый чекаут (блочный) и является вендор-зависимым от темы woodmart. Помимо этого, отключать целиком плагин (включая его регистрацию) является не совсем корректным поведением. С нашей стороны мы не будем добавлять похожий код, поскольку расчет стоимости доставки выполняется только по запросу woocommerce, соответственно если это необходимо отключить в корзине, то менять надо именно вызов, а не наш плагин.
2. Все тексты переведены. В js коде также используется перевод, согласно документации. Если у вас установлены актуальные переводы Wordpress, то строки в JS файлах будут выводится пользователю как переведенные.

**Comment by oleg150587-pixel** [comment]
Здравствуйте! У меня та же проблема с добавлением товара в корзину, по времени 4-5 секунд занимает, это очень долго. Как можно это исправить? Тема стоит woodmart, но код выше не помог ускорить добавление товара в корзину. Помогите пожалуйста решить проблему. Спасибо!

**Reply by theelementwork** [reply]
В файле /wp-content/plugins/cdekdelivery/cdek.php модифицированный код целиком который перестает перегружать корзину на woodmart. 

`defined('ABSPATH') or exit;

if(!((isset($_GET['wc-ajax']) && ($_GET['wc-ajax'] == 'add_to_cart' || $_GET['wc-ajax'] == 'remove_from_cart')) || (isset($_GET['action']) && ($_GET['action'] == 'woodmart_update_cart_item' || $_GET['action'] == 'woodmart_ajax_add_to_cart')) || isset($_GET['key']))){
if (file_exists(__DIR__ . '/vendor/autoload.php')) {
    require __DIR__ . '/vendor/autoload.php';
}

if (!class_exists(\Cdek\Loader::class)) {
    trigger_error('CDEKDelivery not fully installed! Please install with Composer or download full release archive.',
                  E_USER_ERROR);
}

\Cdek\Loader::new()(__FILE__);
}`

Я обычно после каждого обновления плагина опять ставлю эту модификацию с 19 строки и до конца файла целиком и все опять работает нормально.

**Reply by theelementwork** [reply]
[cdek.php](https://github.com/user-attachments/files/22730520/cdek.php)
Вставился код с ошибкой... я прикрепил файл с 19 строки и до конца файла копировать и вставить вместо оригинала.

**Reply by oleg150587-pixel** [reply]
Спасибо! Простые товары стали добавляться в корзину быстрее. Но на вариативные товары это не сработало((( Стоит плагин Variation Swatches for WooCommerce, может еще и в нем проблема. Если у вас есть мысли как это можно исправить, буду сильно благодарен🙏

**Reply by theelementwork** [reply]
> Спасибо! Простые товары стали добавляться в корзину быстрее. Но на вариативные товары это не сработало((( Стоит плагин Variation Swatches for WooCommerce, может еще и в нем проблема. Если у вас есть мысли как это можно исправить, буду сильно благодарен🙏

Я только увидел ваше сообщение. Честно вам скажу я только недавно тоже выбирал нормальный модуль для вариативных товаров и в итоге пришел к выводу что самый лучший модуль это сам встроенный функционал Woodmart. В нем уже прекрасно реализовано все. Но работает правильно, как нам всем нужно, по умолчанию только с Linked Variations. Что я сделал? Дописал дополнительный код с помощью Cloude для functions.php для дочерней темы woodmart-child. Я прикрепил файлик. Я внутри реализовал "Автоматическое включение свотчей для всех PA-глобальных атрибутов" чтобы убрать выпадающий список и заменить правельным функционалом кнопок, картинок и цветов и "Конвертация кастомных атрибутов в глобальные при сохранении продукта" чтобы любые кастомные атрибуты превращались в глобальные и движок их видел. После чего внешний вид вариаций можно настраивать через атрибуты 
<img width="1030" height="1077" alt="image" src="https://github.com/user-attachments/assets/36cb3c2b-5bcd-45f0-8852-0131a7ddda0a" />. Это практически сводит к нулю нагрузку на движок так как будет использоваться родной функционал Woodmart и Woocommerce. При этом результат даже красивее чем в плагине Variation Swatches for WooCommerce. И работает прекрасно. Можно определять как фотографии так и кнопки и цвета, в общем полностью весь функционал как у стороннего плагина. При желании можете этот код реализовать в сниппет в виде плагина. 


[functions.php](https://github.com/user-attachments/files/23188914/functions.php)

**Reply by theelementwork** [reply]
Для выбора атрибута который будет подтягивать картинку товара  нужно выбрать один атрибут в Theme Settings - Shop - Variable Products - Grid swatch attribute to display (выбрать атрибут который будет вариативить как картинку) Как правило это Color (у меня так) или Type, но можно выбрать любой. 

Кнопки он по умолчанию будет делать вместо выпадающего списка. А если вам нужно чтобы цвета делал то в атрибутах каждого цвета нужно указать нужный цвет из палитры или картинки прикреплять вместо цвета, тогда будет ввиде цветов в кружках квадратиках или как вы настроите в настройках конкретного атрибута.
